versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -591,20 +595,32 @@ public PaginationIterator<KeyEntry> getKeys(
             ContentKey key = storeKeyToKey(indexElement.key());
             CommitOp commitOp = indexElement.content();
             Content.Type contentType = contentTypeForPayload(commitOp.payload());
+            Content content = null;
 
-            if (withContent) {
-              Content c =
+            UUID contentId = commitOp.contentId();
+            String contentIdString;
+            if (contentId != null) {
+              contentIdString = contentId.toString();
+            } else {
+              // this should only be hit by imported legacy nessie repos
```

Comment:
this comment is my understanding of why it might be null from [here](https://github.com/projectnessie/nessie/blob/aafa1868db660b397d6eda43c715ca1c0d122692/versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/objtypes/CommitOp.java#L50-L55)